xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-ui/src/XWiki/Lightbox/WebHome.xml

```diff
@@ -0,0 +1,838 @@
+        $('#imageDownload').attr('download', getImageName(image[0].src));
+      });
+    });
+  };
+
+  /**
+   * Remove the popover for the given elements.
+   */
+  var removePopovers = function(elements) {
+    elements.each(function(index) {
+      $(this).popover('destroy');
+    });
+  }
+
+  /*
+   * Extract the image name from the url. This may not be applied for external urls.
+   */
+  var getImageName = function(imageLink) {
+    return new URL(imageLink).pathname.split("/").pop();
+  };
+
+  /**
+   * Rescale image to original size.
+   */
+  var removeResizeParams = function(imageSrc) {
+    var url = new URL(imageSrc);
+    var searchParams = new URLSearchParams(url.search);
+    searchParams.delete('width');
+    searchParams.delete('height');
+    url.search = searchParams.toString();
+    return url.toString();
+  };
+
+  /**
+   * Scale image to thumbnail. This may not be applied for external urls.
+   */
+  var createThumbnailURL = function(imageSrc) {
+    var url = new URL(imageSrc);
+    var searchParams = new URLSearchParams(url.search);
+    searchParams.append('width', '150');
+    searchParams.append('height', '150');
+    url.search = searchParams.toString();
+    return url.toString();
+  };
+
+  /**
+   * Open lightbox in fullscreen mode, or close it if already open.
+   */
+  var toggleLightboxFullscreen = function() {
+    if (!$('#lightboxFullscreen').data('open')) {
+      myOpenLightbox.requestFullScreen($('#blueimp-gallery')[0]);
+      $('#lightboxFullscreen').data('open', true);
+    } else {
+      myOpenLightbox.exitFullScreen();
+      $('#lightboxFullscreen').data('open', false);
+    }
+  };
+
+  /**
+   * Update lightbox description using given information.
+   */
+  var updateLightboxDescription = function(data) {
+    if (data != undefined) {
+      $('.lightboxDescription').css("display", "grid");
+      $('.lightboxDescription').find('.title').text(data.name);
+      var translationPublisher = l10n.get('author', XWiki.Model.resolve(data.author, XWiki.EntityType.DOCUMENT).name);
+      $('.lightboxDescription').find('.publisher').text(translationPublisher);
+      var translationDate = l10n.get('date', new Date(data.date).toLocaleDateString());
+      $('.lightboxDescription').find('.date').text(translationDate);
+    } else {
+      $(".lightboxDescription").css("display", "none");
+    }
+  }
+
+  /**
+   * Get information for attachment, when this can be found inside the current page.
+   * TODO: consider the path of the attachment, not just the current page.
+   */
+  var getAttachmentInfo = function(imageLink) {
+    var deferred = $.Deferred();
+    var attachmentsURL = xm.restURL + `/attachments`;
+    $.ajax(attachmentsURL, {
+      method: 'GET',
+      dataType: 'json'
+    }).done(function (data, textStatus, jqXHR) {
+      var attachment = data.attachments.find(field => field.name == getImageName(imageLink));
+      deferred.resolve(attachment);
```

Comment:
This seems to reload the information about all attachments of the page for every viewed image which seems a waste of resources that could also lead to delays. I think it would be better to cache the result of the request, even when actually querying the attachments of other pages this would still be a good idea.

Reply:
We could also get directly the meta data for a specific image attachment (based on the file name). But even in this case the meta data should be cached and not requested each time the same image attachment is viewed in the lightbox.